util/unifycr/src/unifycr.c

```diff
@@ -0,0 +1,269 @@
+
+        case 'o':
+            transfer_out = strdup(optarg);
+            break;
+
+        case 'h':
+        default:
+            usage(0);
+            break;
+        }
+    }
+
+    opts_cmd.cleanup = cleanup;
+    opts_cmd.consistency = consistency;
+    opts_cmd.mountpoint = mountpoint;
+    opts_cmd.transfer_in = transfer_in;
+    opts_cmd.transfer_out = transfer_out;
+}
+
+int main(int argc, char **argv)
+{
+    int i = 0;
+    int ret = 0;
+    char *cmd = NULL;
+
+    program = basename(argv[0]);
+
+    if (argc < 2)
+        usage(1);
+
+    cmd = argv[1];
+
+    for (i = 0; i < sizeof(actions)/sizeof(char *); i++) {
+        if (strncmp(cmd, actions[i], strlen(actions[i])) == 0)
+            action = i;
+    }
+
+    if (action < 0)
+        usage(1);
+
+    parse_cmd_arguments(argc, argv);
+
+    if (debug) {
+        printf("\n## options from the command line ##\n");
+        printf("cleanup:\t%d\n", opts_cmd.cleanup);
+        printf("consistency:\t%s\n",
+               unifycr_write_consistency(opts_cmd.consistency));
+        printf("mountpoint:\t%s\n", opts_cmd.mountpoint);
+        printf("transfer_in:\t%s\n", opts_cmd.transfer_in);
+        printf("transfer_out:\t%s\n", opts_cmd.transfer_out);
+    }
+
+    ret = unifycr_read_resource(&resource);
+    if (ret) {
+        fprintf(stderr, "failed to detect a resource manager\n");
+        goto out;
+    }
+
+    if (debug) {
+        printf("\n## job allocation (%llu nodes) ##\n",
+               _llu(resource.n_nodes));
+        for (i = 0; i < resource.n_nodes; i++)
+            printf("%s\n", resource.nodes[i]);
+    }
+
+    ret = unifycr_read_env(&opts_env);
+    if (ret) {
+        fprintf(stderr, "failed to environment variables\n");
+        goto out;
+    }
+
+    if (debug) {
+        printf("\n## options from the environmental variables ##\n");
+        printf("UNIFYCR_MT:\t%s\n", opts_env.unifycr_mt);
+    }
+
+    ret = unifycr_read_sysconf(&opts_sysconf);
+    if (ret) {
+        fprintf(stderr, "failed to read config file\n");
+        goto out;
+    }
+
+    if (debug) {
+        printf("\n## options read from %s ##\n", CONFDIR "/unifycr.conf");
+        printf("runstatedir:\t%s\n", opts_sysconf.runstatedir);
+        printf("consistency:\t%s\n",
+               unifycr_write_consistency(opts_sysconf.consistency));
+        printf("mountpoint:\t%s\n", opts_sysconf.mountpoint);
+    }
+
+    ret = unifycr_write_runstate(&resource, &opts_sysconf, &opts_env,
+                                 &opts_cmd);
+    if (ret) {
+        fprintf(stderr, "failed to write the runstate file\n");
+        goto out;
+    }
+
+    if (debug) {
+        int tmp;
+        char cmdbuf[1024] = { 0, };
+
+        printf("\n## the final runstate file %s/unifycr-runstate.conf ##\n",
+               opts_sysconf.runstatedir);
+
+        sprintf(cmdbuf, "cat %s/unifycr-runstate.conf",
+                        opts_sysconf.runstatedir);
+        tmp = system(cmdbuf);
```

Comment:
I think it would be better to open and print the file contents directly, rather than calling `cat`.